include/aspect/geometry_model/ellipsoidal_chunk.h

```diff
@@ -0,0 +1,268 @@
+         */
+        virtual std::set<types::boundary_id>
+        get_used_boundary_indicators() const;
+
+        /*
+        *Set symbolic names for boudaries (mrtf)
+        */
+        virtual std::map<std::string,types::boundary_id>
+        get_symbolic_boundary_names_map() const;
+
+        /**
+         * Declare the parameters this class takes through input files.
+         */
+        static
+        void
+        declare_parameters(ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter
+         * file.
+         */
+        virtual
+        void
+        parse_parameters(ParameterHandler &prm);
+
+        /**
+         * Calculate radius at current position.
+         */
+        double
+        get_radius(const Point<dim> &point) const;
+
+        /**
+         * Retrieve the semi minor axis b value.
+         */
+        double
+        get_semi_minor_axis_b() const;
+
+        /**
+         * Retrieve the semi major axis a value.
+         */
+        double
+        get_semi_major_axis_a() const;
+
+
+        /**
+         * Retrieve the value of the eccentricity.
+         */
+        double
+        get_eccentricity() const;
+
+
+        /**
+         * Retrieve the rot_para_to_para_angle variable.
+         */
+        double
+        get_rot_para_to_para_angle() const;
+
+
+        /**
+         * Retrieve the para_to_rect_angle variable.
+         */
+        double
+        get_para_to_rect_angle() const;
+
+
+        /**
+         * Retrieve the rotation_longitude variable.
+         */
+        double
+        get_rotation_longitude() const;
+
+
+        /**
+         * Retrieve the rotation_latitude variable.
+         */
+        double
+        get_rotation_latitude() const;
+
+
+        /**
+         * Retrieve the manifold object.
+         */
+        EllipsoidalChunkGeometry
+        get_manifold() const;
+
+      private:
+        // Declare variables for reading in coordinates of the region of interest.
+        std::vector<std::vector<double> > corners;
+        std::vector<std::vector<double> > corners_parallelogram;
+        std::vector<std::vector<double> > corners_rectangle;
+        double semi_major_axis_a;
+        double eccentricity;
+        double semi_minor_axis_b;
+        double rot_para_to_para_angle;
+        double para_to_rect_angle;
+        double rotation_longitude;
+        double rotation_latitude;
+        double bottom_depth;
+        double westLongitude;
+        double eastLongitude;
+        double northLatitude;
+        double southLatitude;
+        // Declare variables for subdividing
+        int EW_subdiv;
+        int NS_subdiv;
+        int depth_subdiv;
+
+
+
+        /**
+         * Construct manifold object Pointer to an object that describes the geometry.
+         */
+        EllipsoidalChunkGeometry   manifold;
+
+        static void set_manifold_ids (Triangulation<dim> &triangulation)
+        {
```

Comment:
Can you move the implementation of this and the next function to the `.cc` file?